### PR TITLE
Add DASH API

### PR DESCRIPTION
**Pull request by marian-pritsak**
Includes:
API for VNET to VNET use case
API for virtual port aka ENI
API for the user ACL

Signed-off-by: Marian Pritsak <marianp@mellanox.com>

### CODE REVIEW

**Review by kcudnik — APPROVED** [review]

**Comment by prsunny** [comment]
@marian-pritsak , would you please sync this to latest dash branch?

**Comment by marian-pritsak** [comment]
Done, but depends on a fix from https://github.com/opencomputeproject/SAI/pull/1643

**Review by lguohan — APPROVED** [review]

File: experimental/saiexperimentaldashoutboundrouting.h [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+ */
+typedef enum _sai_outbound_routing_entry_action_t
+{
+    SAI_OUTBOUND_ROUTING_ENTRY_ACTION_ROUTE_VNET,
+
+    SAI_OUTBOUND_ROUTING_ENTRY_ACTION_ROUTE_VNET_DIRECT,
+
+    SAI_OUTBOUND_ROUTING_ENTRY_ACTION_ROUTE_DIRECT,
+
+    SAI_OUTBOUND_ROUTING_ENTRY_ACTION_DROP,
```
Comment (reshmaintel) [inline experimental/saiexperimentaldashoutboundrouting.h:47]: The DROP action is not needed for direction, and should be removed.

**Comment by vmytnykx** [comment]
Regenerated APIs by using sai_api_gen.py, and looks like this branch needs to be rebased and updated according to latest DASH API pipeline. Once rebased, the DENY also will be removed automatically.

Regards,
Volodymyr

File: experimental/saiexperimentaldashoutboundrouting.h (hunk 1 above)
Comment (reshmaintel) [reply experimental/saiexperimentaldashoutboundrouting.h:47]: @marian-pritsak  Hi Marian, Could you please re generate the DASH SAI APIs based on the current P4. Thanks

Comment (marian-pritsak) [reply experimental/saiexperimentaldashoutboundrouting.h:47]: Done

**Review by reshmaintel — APPROVED** [review]

**Pull request merged**